python/lsst/ts/Dome/dome_csc.py

```diff
@@ -405,26 +424,48 @@ async def inflate(self, data):
 
         Parameters
         ----------
-        data : `TBD`
-            The contents of this parameter will be defined soon.
+        data : `dict`
+            A dictinary with arguments to the function call. It should contain the key "action" with a
+            string value (ON or OFF).
         """
-        raise salobj.ExpectedError("Not implemented")
+        cmd = {"inflate": {"action": data["action"]}}
+        await self.write_then_read_reply(cmd)
 
     async def status(self):
         """Status command not to be executed by SAL.
 
         This command will be used to request the full status of all lower level components.
         """
         cmd = {"status": {}}
-        await self.write(cmd)
-        self.lower_level_status = await self.read()
+        self.lower_level_status = await self.write_then_read_reply(cmd)
+        self.log.info(self.lower_level_status)
 
         # Remove some keys because they are not reported in the telemetry.
-        amcs_keys_to_remove = {"status"}
-        telemetry = self.remove_keys_from_dict(
-            self.lower_level_status["AMCS"], amcs_keys_to_remove
+        keys_to_remove = {"status"}
+        adb_telemetry = self.remove_keys_from_dict(
+            self.lower_level_status["AMCS"], keys_to_remove
+        )
+        self.tel_domeADB_status.set_put(**adb_telemetry)
+        aps_telemetry = self.remove_keys_from_dict(
+            self.lower_level_status["ApSCS"], keys_to_remove
+        )
+        self.tel_domeAPS_status.set_put(**aps_telemetry)
+        l_telemetry = self.remove_keys_from_dict(
+            self.lower_level_status["LCS"], keys_to_remove
+        )
+        self.tel_domeLouvers_status.set_put(**l_telemetry)
+        lws_telemetry = self.remove_keys_from_dict(
+            self.lower_level_status["LWSCS"], keys_to_remove
+        )
+        self.tel_domeLWS_status.set_put(**lws_telemetry)
+        mon_telemetry = self.remove_keys_from_dict(
+            self.lower_level_status["MonCS"], keys_to_remove
+        )
+        self.tel_domeMONCS_status.set_put(**mon_telemetry)
+        th_telemetry = self.remove_keys_from_dict(
+            self.lower_level_status["ThCS"], keys_to_remove
```

Comment:
To reduce boilerplate, please consider a method that removes the keys and puts the remaining data. It could take a topic (such as `self.tel_domeADB_status`) and the data dict.

I have no opinion whether to pass in the whole data dict or just the portion that is relevant to the topic (if the former you would presumably look up the key by topic name).

Reply:
I **knew** that you were going to comment that. I planned on doing that and unfortunately it slipped my mind. Thanks for the reminder.